BridgeSDK/BridgeAPI/SBBUploadManager.m

```diff
@@ -290,19 +346,32 @@ - (void)toldBridgeFileUploadedWithTask:(NSURLSessionUploadTask *)uploadTask forB
 - (void)handleUploadHTTPStatusCode:(NSInteger)httpStatusCode forUploadFilePath:(NSString *)filePath
 {
     switch (httpStatusCode) {
-        case 403: {
-            // this will have happened because the pre-signed url timed out before starting the actual upload to S3,
-            // so try, try again
+        case 403:
+        case 500: {
+            // 403 for our purposes means the pre-signed url timed out before starting the actual upload to S3.
+            // 500 means internal server error ("We encountered an internal error. Please try again.")
+            // either way we should try again immediately
 #if DEBUG
-            NSLog(@"Background file upload to S3 failed with HTTP status 403 (presumably due to presigned url timeout); retrying...");
+            NSLog(@"Background file upload to S3 failed with HTTP status %ld; retrying...", (long)httpStatusCode);
 #endif
             NSDictionary *uploadRequestJSON = [self uploadRequestJSONForFile:filePath];
             [self kickOffUploadForFile:filePath uploadRequestJSON:uploadRequestJSON];
         } break;
             
+        case 503: {
+            // 503 means service not available or the requests are coming too fast, so try again after
+            // at least a minimum delay
+            NSTimeInterval delay = 5. * 60.; // at least 5 minutes, actually whenever we get to it after that
+            
+#if DEBUG
+            NSLog(@"Background file upload to S3 failed with HTTP status 503; will retry some time after %ld seconds have elapsed.", (long)delay);
+#endif
+            [self setRetryTime:[NSDate dateWithTimeIntervalSinceNow:delay] forFile:filePath];
+        } break;
+            
         default: {
             // iOS handles redirects automatically so only e.g. 307 resource not changed etc. from the 300 range should end up here
-            // (along with all 4xx and 5xx of course)
+            // (along with all unhandled 4xx and 5xx of course)
```

Comment:
Is there any reason we should get unrecoverable errors? Would an assert here be helpful?